### DISCUSSION TITLE
[Feature Request] Usage of dovecot expire plugin

### DISCUSSION THREAD

**Discussion by piwats** [modoboa/modoboa#571] [Ideas]
Hello there,

since modoboa can use db for storing much of the user related settings, wouldn't it be nice to implement the use of the dovecot expire plugin which would allow the user to expunge a definden folder after some time? F. e. delete all spam mail in the spam folder after 14d?

--> http://wiki2.dovecot.org/Plugins/Expire

-Paul

**Comment by tonioo** [comment]
It might be a nice feature to add.

I don't use the expire plugin so I don't know how I would integrate it within modoboa. Can you provide a more detailled example ?

**Comment by piwats** [comment]
As far as I understand, the expire plugin can be configured via the dict service of dovecot to look in a db like this (from the dovecot wiki):

CREATE TABLE expires (
  username varchar(75) not null,
  mailbox varchar(255) not null,
  expire_stamp integer not null,
  primary key (username, mailbox)
);

There we have all obvious neede settings to us stored.
Then dovecote is setup to use this db as well described in the wiki:

connect = host=localhost dbname=mails user=sqluser password=sqlpass

map {
  pattern = shared/expire/$user/$mailbox
  table = expires
  value_field = expire_stamp

  fields {
    username = $user
    mailbox = $mailbox
  }
}

That should be also understandable, I hope. Also, I hope that I understand the dovecot wiki entry about this properly.

After the expire plugin is enabled in dovecot, it should be able, as stated in the wiki, to get the needed information about which mailbox is to be expunged and from which date on.

As I understand, modoboa can write and read settings from db - the other thing would be to create a user interface.

Unfortunately I am not a developer and my skills are pretty limited - coding is, to my shame, not a skill I can offer :(.

**Comment by tonioo** [comment]
Actually the expire plugin is just a way to optimize expunging of old mails.

The expunging operation is not automatic, you need to run doveadm. For example:

```
doveadm expunge -A mailbox Trash savedbefore 30d
```

As the wiki says: **This command runs fine even without the expire plugin**. 

If the plugin is enabled and a SQL dict is configured, dovecot will use a table to store per-mailbox information but this table is not intended to be modified from the outside.

As far as I understand, you can't configure the plugin behaviour once it is activated (from dovecot configuration files). Eventually, this feature is not as interesting as it seemed. Modoboa could possibly create a compatible table...

The remaining part is just a matter of configuration (cron job for example).

What do you think?

**Comment by piwats** [comment]
Would it be then possible via the manage.py interface to create an additional command, that would gather all required data (user, mailboxes to expunge, oder then & co.), for example which folder is to be expunged, and which messages have to be deleted older then a time, from the modoboa db? ---> cron?

The easiest implementation is the way to go I think.
